### DISCUSSION TITLE
Display the relationship in the view?

### DISCUSSION THREAD

**Discussion by pablovt** [filamentphp/filament#8242] [General]
I'm having difficulty presenting the relationship in the general listing. I have the same problem as this person reported here:
 [https://stackoverflow.com/questions/76959230/laravel-filamentphp-3-0-relation-column](url)

**Comment by leandrocfe** [comment]
what is the issue?

**Reply by pablovt** [reply]
I have a table called device and another service. I called the service with the device, by foreign key. When creating a new device, it lists the registered services, but it does not appear in the list. What should I do to show it, since I followed the documentation orienting to perform `servico.name` and it doesn't appear.

**Reply by leandrocfe** [reply]
Share your code and your models please

**Reply by pablovt** [reply]
Yes. My model device:
```
class Device extends Model
{
    use HasFactory;

    protected $fillable = ['name'];

    public function services(): BelongsTo
    {
        return $this->belongsTo(Service::class);
    }
}
```

And model Service:

```
class Service extends Model
{
    use HasFactory;

    protected $fillable = ['name'];

    public function device(): HasMany
    {
        return $this->hasMany(Device::class);
    }
}

```
And my DeviceResource:
```
return $table
            ->columns([
                Tables\Columns\TextColumn::make('service.name'), //Our 'services.name' does not work
                Tables\Columns\TextColumn::make('created_at')
                    ->dateTime()
                    ->sortable()
                    ->toggleable(isToggledHiddenByDefault: true),
                Tables\Columns\TextColumn::make('updated_at')
                    ->dateTime()
                    ->sortable()
                    ->toggleable(isToggledHiddenByDefault: true),
            ])
```

**Reply by leandrocfe** [reply]
Are you sure that your relationship is working?

**Reply by pablovt** [reply]
In my form, is working. Only in the list does not work.

**Reply by pablovt** [reply]
In DeviceResource form:
```
Select::make('service_id')
                    ->relationship('services', 'name')
                    ->preload(),
```

**Reply by leandrocfe** [reply]
Weird, but you have 'nome' in fillable, 'name' as column. This should be `devices()` and `service()`, following the laravel pattern.
You should check the laravel docs.

**Reply by pablovt** [reply]
Sorry, that correct name is 'name'. What is the explanation of working only in the form?

**Reply by pablovt** [reply]
I identified my mistake. It was in the name that I created the class. Thank you for the tips.

**Comment by pablovt** [comment]
Resolved.
